src/app/pages/applications/chart-releases/chart-releases.component.html

```diff
@@ -26,9 +26,13 @@
         </div>
 
         <div class="chart-button-row" fxLayout="row" fxLayoutAlign="end end" (click)="$event.stopPropagation()">
-          <button mat-button (click)="portal(item)" *ngIf="item.status === ChartReleaseStatus.Active && item.portals && item.portals.web_portal && item.portals.web_portal[0]"
-              ix-auto ix-auto-type="button" ix-auto-identifier="{{ item.name}}-portal">
-            {{ 'Portal' | translate }}
+          <button mat-button (click)="userPortal(item)" *ngIf="item.status === ChartReleaseStatus.Active && item.portals && item.portals.user_portal && item.portals.user_portal[0]"
```

Comment:
> Is it up to every app catalog to define names for portals or is it driven by truenas middleware?

It's set in questions.yaml, where portals gets loaded into item.portals.

Let me show a snippet:
```
portals:
  web_portal:
    protocols:
      - "$kubernetes-resource_configmap_portal_protocol"
    host:
      - "$kubernetes-resource_configmap_portal_host"
    ports:
      - "$kubernetes-resource_configmap_portal_port"
```

> If we add support for admin portals now, how would it work with an official app catalog?

Currently buttons are not shown if either portals as a whole or the specific portal are not set.
So in the case of the official catalog it would just not show the admin portal button :)

> In either case, I'd prefer to see changes made elsewhere before seeing them in webui purely so that I could test new behaviour. 

I'll work on setting up a demo to test and showcase this :)

> Having backward compatibility would also be nice, but it depends on answers to previous two questions.

Thinking about this some more, and lets not rename web_portal for compatibility at all, would make it much easier and less breaking to implement... I'll add that later.

Reply:
Would it make sense to somehow add support for portal label in `questions.yaml` and let developers have whatever portal buttons they want?

Reply:
I think that would be better suited for a seperate PR, as this is mostly focussed on splitting user-facing portal(s) from admin-only portal(s), which is a quite frequent occurance.

As those portals are lists, it's quite easy to, in the future, add customizable labels. Though i'm not sure it would require a middleware change to do so. Might be more suited for Bluefin tbh.

Reply:
Demo is working:
![image](https://user-images.githubusercontent.com/7613738/133588921-8e6aef06-69b5-4d3a-93fe-52d21d990776.png)

Reply:
I'll now take a look at making dynamic labels :)

Reply:
Okey, i've been able to implement dynamic labels based on the name of the actual portals inside questions.yaml :)
In the TrueCharts catalog, in the "incubator" train there is an example App based on pihole available to test :)

Known downsides:
Needs work oin the ports, possibly move those and give Catalog creators that whole bare to work with.

Reply:
Okey @undsoft :
Dynamic naming and the number of buttons depending on the questions.yaml file is done.
It also dynamically scales the ports now and hides the port references completely if they become to small.

It think that's all you feedback done :)